### PR TITLE
pangea-node-sdk: add Audit export API (GEA-13176)

### PR DESCRIPTION
**Pull request by kenany**

### CODE REVIEW

File: packages/pangea-node-sdk/src/errors.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export namespace PangeaErrors {
       this.response = response;
     }
 
-    get pangeaResponse(): any {
+    get pangeaResponse(): PangeaResponse<Errors> {
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/errors.ts:68]: I'm not sure if this is not an issue with 202 responses where `result` have `accepted_result` and other info

File: packages/pangea-node-sdk/src/services/audit.ts [hunk 2]
```diff
@@ -190,7 +190,8 @@ class AuditService extends BaseService {
       response = await this.post("v2/log_async", data, postOptions);
     } catch (e) {
       if (e instanceof PangeaErrors.AcceptedRequestException) {
-        return e.pangeaResponse;
+        // TODO: bad type cast
+        return e.pangeaResponse as unknown as PangeaResponse<Audit.LogBulkResponse>;
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/services/audit.ts:194]: Okey, I guess you fixed above comment with this

File: packages/pangea-node-sdk/src/services/audit.ts [hunk 3]
```diff
@@ -450,6 +490,10 @@ class AuditService extends BaseService {
   downloadResults(
     request: Audit.DownloadRequest
   ): Promise<PangeaResponse<Audit.DownloadResult>> {
+    if (!request.request_id && !request.result_id) {
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/services/audit.ts:493]: What if they set both?... check or not check (in SDKs), that's the question...

File: packages/pangea-node-sdk/src/types.ts [hunk 4]
```diff
@@ -223,26 +223,21 @@ export namespace Audit {
...
-     * ID returned by the search API.
-     */
-    result_id: string;
+    /** ID returned by the search API. */
+    request_id?: string;
 
-    /**
-     * Format for the records.
-     */
+    /** ID returned by the search API. */
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/types.ts:237]: Same comment than in python https://github.com/pangeacyber/pangea-python/pull/337#discussion_r1588260992

File: packages/pangea-node-sdk/tests/integration/audit.test.ts [hunk 5]
```diff
@@ -1041,3 +1041,35 @@ it("log stream", async () => {
...
+it("export download", async () => {
+  const exportRes = await auditGeneral.export({ verbose: false });
+  expect(exportRes.status).toStrictEqual("Accepted");
+
+  // Note that the export can easily take dozens of minutes, if not longer, so
+  // we don't actually wait for the results on CI. Instead we just poll it once
+  // and then attempt the download, even when we know it isn't ready yet, just
+  // to verify that the core of the functions are working.
+  try {
+    await auditGeneral.pollResult(exportRes.request_id);
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/tests/integration/audit.test.ts:1054]: Retry logic?

**Review by pangea-andrest — APPROVED** [review]

File: packages/pangea-node-sdk/src/types.ts (hunk 4 above)
Comment (kenany) [reply packages/pangea-node-sdk/src/types.ts:237]: Corrected now.

File: packages/pangea-node-sdk/tests/integration/audit.test.ts (hunk 5 above)
Comment (kenany) [reply packages/pangea-node-sdk/tests/integration/audit.test.ts]: Done.

**Review by pangea-andrest — APPROVED** [review]

**Pull request merged**